crates/maybenot/src/framework.rs

```diff
@@ -205,6 +208,30 @@ where
         self.current_time = current_time;
         for e in events {
             self.process_event(e);
+
+            // handle internal signaling
+            // Special case: if a machine sends a signal which results in another
+            // signal in response, that response may or may not be received; this
+            // depends on the order in which machines are given to the framework.
+            // Check for this explicitly (keep track of excluded machines).
+            if let Some(signal) = self.signal_pending {
+                let mut excluded = None;
+
+                for mi in 0..self.runtime.len() {
+                    if signal.is_some() && signal.unwrap() == mi {
```

Comment:
Both to make it simpler, and to avoid the (only theoretical) panic in the `unwrap` can you try `if signal.take() == Some(mi)`?

Reply:
Code refactored, worked in.